src/client/datascience/raw-kernel/liveshare/hostRawNotebookProvider.ts

```diff
@@ -233,7 +234,7 @@ export class HostRawNotebookProvider
                 !kernelConnectionMetadata ||
                 (kernelConnectionMetadata?.kind === 'startUsingKernelSpec' && !kernelConnectionMetadata?.kernelSpec)
             ) {
-                notebookPromise.reject('Failed to find a kernelspec to use for ipykernel launch');
+                notebookPromise.reject(new KernelSpecNotFoundError());
```

Comment:
I believe this is one of the culptrits.
I can see that we attempt to start a notebook, & then we don't even attempt to start the raw kernel, but it just dies prematurely & we have absolutely no idea.

If this is the cause, then we know exactly what to fix. Because rejecting with a string, the telemetry cannot determine the cause of the failure & nothing is logged.